Dockerfile

```diff
@@ -49,23 +40,33 @@ RUN apt-get install -y \
     libxml2-dev \
     libxslt-dev
 
-# APP directory setup
-RUN adduser --system --disabled-login docker \
- && mkdir -p "$BASE_DIR" "$APP_DIR" "$APP_DIR/src/assets" "$APP_DIR/src/media" \
- && chown -R docker:nogroup "$BASE_DIR" "$APP_DIR"
-
 # Install Poetry
 RUN pip install --no-cache-dir pip==23.3.2 && \
     pip install --no-cache-dir poetry==1.8.2
 
-# Install Python dependencies (only main dependencies)
-COPY --chown=docker:nogroup pyproject.toml poetry.lock ./
+# Install Python dependencies
+COPY pyproject.toml poetry.lock ./
 RUN poetry install --no-root && \
```

Comment:
We could
1. Use and change Dockerfile like following
```
ARG TARGET
RUN if [ "$TARGET" = "prod" ]; then \
        poetry install --no-root; \
        yes | poetry cache clear --all pypi; \
    else \
        poetry install --only main --no-root; \
        yes | poetry cache clear --all pypi; \
    fi
```

or

2. We could change to
```
        ## python_dependencies
        poetry install --no-root --only main; \
        yes | poetry cache clear --all pypi; \
```
And add a line in dev stage to install dev dependencies
```
       ## dev
       poetry install --no-root --only dev;
```

What do you think?

Reply:
I prefer method 2 since debugging for conditional statements in Dockerfile is extremely painful, so I've never written `if...else` in the Dockerfile.